### PR TITLE
Add http/2 and gRPC test server code as new subcommands

### PR DESCRIPTION
**Pull request by frobware**
This adds the http/2 and gRPC test server code from origin/test/extended/router.
This allows us to reference the test server code in origin tests using the cluster-ingress-operator image.

### CODE REVIEW

**Comment by frobware** [comment]
/hold

Testing integration with origin tests.

File: go.mod [hunk 1]
```diff
@@ -39,3 +39,5 @@ require (
 	sigs.k8s.io/controller-runtime v0.6.4
 	sigs.k8s.io/controller-tools v0.4.1
 )
+
+require google.golang.org/grpc v1.27.1
```
Comment (sgreene570) [inline go.mod:43]: Any reason not to put this in the main require block with a comment explaining why it is needed?
Or is this just go.mod being sticky?

File: cmd/ingress-operator/http2-testserver.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package main
+
+import (
+	http2testserver "github.com/openshift/cluster-ingress-operator/test/http2"
+	"github.com/spf13/cobra"
+)
+
+func NewServeHTTP2TestServerCommand() *cobra.Command {
+	var command = &cobra.Command{
+		Use:   "serve-http2-testserver",
```
Comment (sgreene570) [inline cmd/ingress-operator/http2-testserver.go:10]:
```suggestion
		Use:   "serve-http2-test-server",
```
testserver should be test-server probably (not just here)?

File: test/grpc/server/server.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
+package server
```
Comment (sgreene570) [inline test/grpc/server/server.go:1]: The directory / package of this file do no match the pattern used for `test/http2/server.go`.

File: test/http2/server.go [hunk 4]
```diff
@@ -0,0 +1,55 @@
+package http2
```
Comment (sgreene570) [inline test/http2/server.go:1]: Should we also move the canary server to `test/http/serve-healthcheck.go` to math these 2 new files?
IMO splitting between `cmd/` and `test/` is a little confusing.

Comment (sgreene570) [reply test/http2/server.go:1]: Maybe we could also combine all 3 of the cobra command functions into one file in `cmd/` called something along the lines of `serve-test-commands.go`.

File: test/grpc/client/client.go [hunk 5]
```diff
@@ -0,0 +1,145 @@
+package main
```
Comment (sgreene570) [inline test/grpc/client/client.go:1]: Any plans to call this code?

**Review by sgreene570 — COMMENTED** [review]
some small organization questions.

File: go.mod (hunk 1 above)
Comment (frobware) [reply go.mod]: Nope; I just hardcoded for my first pass. Already gone.

File: test/http2/server.go (hunk 4 above)
Comment (frobware) [reply test/http2/server.go:1]: Done in https://github.com/openshift/cluster-ingress-operator/pull/585/commits/489535824d7cf3e70a5c641e48012e589542a725

**Comment by sgreene570** [comment]
New changes look good
/lgtm
/hold cancel

Let's unblock http/2 tests in origin

**Comment by sgreene570** [comment]
```
level=error msg=Error: Error creating VPC: VpcLimitExceeded: The maximum number of VPCs has been reached.
level=error msg=	status code: 400, request id: bdf5c801-8cb2-48fb-86ba-008e54f4e588
```

:desktop_computer:

**Pull request merged**